tools/src/main/java/org/wildfly/transformer/tool/cmdline/Main.java

```diff
@@ -28,35 +28,46 @@
  */
 public final class Main extends Common {
 
+    private static final String PACKAGES_MAPPING_OPTION = "--packages-mapping=";
+
     public static void main(final String... args) throws IOException {
         if (!validParameters(args)) {
             printUsage();
             System.exit(1);
         }
 
-        final File sourceFile = new File(args[0]);
-        final File targetFile = new File(args[1]);
+        final String packagesMappingFile = args.length == 3 ? args[0].substring(PACKAGES_MAPPING_OPTION.length()) : null;
+        final File sourceFile = new File(args.length == 3 ? args[1] : args[0]);
+        final File targetFile = new File(args.length == 3 ? args[2] : args[1]);
         if (sourceFile.getName().endsWith(CLASS_FILE_EXT)) {
-            transformClassFile(sourceFile, targetFile);
+            transformClassFile(sourceFile, targetFile, packagesMappingFile);
         } else if (sourceFile.getName().endsWith(JAR_FILE_EXT)) {
-            transformJarFile(sourceFile, targetFile);
+            transformJarFile(sourceFile, targetFile, packagesMappingFile);
         }
     }
 
     private static boolean validParameters(final String... args) {
-        if (args.length != 2) {
-            System.err.println("2 arguments required");
+        if (args == null || args.length < 2 || args.length > 3) {
+            System.err.println("At least 2 arguments are required");
```

Comment:
Message say that 2 arguments are required.

Reply:
In a future change, we could also support one argument as this bash script logic does by defaulting the second argument to argv[1] jar name + ".output":
```
java -cp $TRANSFORMER_CLASSPATH:/home/smarlow/jars/asm-7.1.jar org.wildfly.transformer.tool.cmdline.Main $jarname $jarname.output

if [ -f "$jarname.output" ]; then
    mv $jarname $jarname.orig
    mv $jarname.output $jarname
else     
    echo "missing $jarname.output"
    exit 2
fi
```